aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -185,13 +185,25 @@ void TensorIterator::allocate_outputs() {
       AT_ASSERTM(op.is_type_defined(), "no type for operand", i);
       int element_size = elementSize(op.dtype);
       op.stride_bytes = compatible_stride(element_size);
-
       auto tensor_shape = invert_perm(shape_);
       auto tensor_stride = invert_perm(op.stride_bytes);
       for (int dim = 0; dim < ndim(); dim++) {
         tensor_stride[dim] /= element_size;
       }
       op.tensor = at::empty_strided(tensor_shape, tensor_stride, op.options());
+      if (input_had_channels_last_format_) {
+          op.tensor.maybe_as_channels_last();
+      }
+    }
+  }
+}
+
+void TensorIterator::analyze_memory_formats() {
+  for (auto& op : operands_) {
+    if (!op.tensor.defined())
+      continue;
+    if (op.tensor.is_contiguous(MemoryFormat::ChannelsLast)) {
+      input_had_channels_last_format_ = true;
```

Comment:
Break here, if the OR relation is what you wanted

Reply:
can we document what the behavior is somewhere?  It's if one of the inputs is ChannelsLast, the returns are channels last?

Also, can't some of the operands be outputs?